request-signature-v2.go

```diff
@@ -174,22 +168,53 @@ func signV2(req http.Request, accessKeyID, secretAccessKey string) *http.Request
 // StringToSign = HTTP-Verb + "\n" +
 // 	 Content-Md5 + "\n" +
 //	 Content-Type + "\n" +
-//	 Date + "\n" +
+//	 Expires + "\n" +
 //	 CanonicalizedProtocolHeaders +
 //	 CanonicalizedResource;
 func getStringToSignV2(req http.Request) string {
 	buf := new(bytes.Buffer)
 	// Write standard headers.
-	writeDefaultHeaders(buf, req)
+	writeSignV2Headers(buf, req)
+	// Write canonicalized protocol headers if any.
+	writeCanonicalizedHeaders(buf, req)
+	// Write canonicalized Query resources if any.
+	writeCanonicalizedResource(buf, req)
+	return buf.String()
+}
+
+// From the Amazon docs:
+//
+// StringToSign = HTTP-Verb + "\n" +
+// 	 Content-Md5 + "\n" +
+//	 Content-Type + "\n" +
+//	 Date + "\n" +
+//	 CanonicalizedProtocolHeaders +
+//	 CanonicalizedResource;
+func getStringToPreSignV2(req http.Request) string {
+	buf := new(bytes.Buffer)
+	// Write standard headers.
+	writePreSignV2Headers(buf, req)
 	// Write canonicalized protocol headers if any.
 	writeCanonicalizedHeaders(buf, req)
 	// Write canonicalized Query resources if any.
 	writeCanonicalizedResource(buf, req)
 	return buf.String()
 }
 
-// writeDefaultHeader - write all default necessary headers
-func writeDefaultHeaders(buf *bytes.Buffer, req http.Request) {
+// writePreSignV2Headers - write preSign v2 required headers.
+func writePreSignV2Headers(buf *bytes.Buffer, req http.Request) {
+	buf.WriteString(req.Method)
```

Comment:
buf.WriteString(req.Method + '\n')

slightly more efficient and fewer lines of code.